src/tm_devices/helpers/constants_and_dataclasses.py

```diff
@@ -219,31 +226,29 @@ def __post_init__(self) -> None:  # noqa: PLR0912,C901,PLR0915
             # this is from an invalid enum name
             raise TypeError(*error.args)  # noqa: TRY200,B904
 
-        # Set the GPIB board number to 0 if it is not provided
-        if self.connection_type == ConnectionTypes.GPIB:
-            if self.gpib_board_number is None:
-                object.__setattr__(self, "gpib_board_number", 0)
-            if not isinstance(self.gpib_board_number, int):
```

Comment:
Just add this condition here and leave the code as-is:

`if not isinstance(self.gpib_board_number, int) and self.gpib_board_number is not None:`